pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/FuncWithAllOptionalInputs.cs

```diff
@@ -6,6 +6,7 @@
 using System.Collections.Immutable;
 using System.Threading.Tasks;
 using Pulumi.Serialization;
+using Pulumi.Example;
```

Comment:
This `using` statement shouldn't be necessary, since all the uses below are fully qualified.

Reply:
I don't understand exactly why it is necessary, but it is. Removing the `using Pulumi.Example` causes the build to fail with:
```
❯ dotnet build -nologo
  Determining projects to restore...
  All projects are up-to-date for restore.
/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/FuncWithAllOptionalInputs.cs(19,177): error CS1061: 'InvokeOptions' does not contain a definition for 'WithVersion' and no accessible extension method 'WithVersion' accepting a first argument of type 'InvokeOptions' could be found (are you missing a using directive or an assembly reference?) [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]
/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/FuncWithAllOptionalInputs.cs(19,169): warning CS8602: Dereference of a possibly null reference. [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]
/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/FuncWithAllOptionalInputs.cs(25,178): error CS1061: 'InvokeOptions' does not contain a definition for 'WithVersion' and no accessible extension method 'WithVersion' accepting a first argument of type 'InvokeOptions' could be found (are you missing a using directive or an assembly reference?) [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]
/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/FuncWithAllOptionalInputs.cs(25,170): warning CS8602: Dereference of a possibly null reference. [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]
/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Inputs/HelmReleaseSettingsArgs.cs(39,27): warning CS8604: Possible null reference argument for parameter 'value' in 'Input<string>.implicit operator Input<string>(string value)'. [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]

Build FAILED.

/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/FuncWithAllOptionalInputs.cs(19,169): warning CS8602: Dereference of a possibly null reference. [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]
/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/FuncWithAllOptionalInputs.cs(25,170): warning CS8602: Dereference of a possibly null reference. [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]
/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Inputs/HelmReleaseSettingsArgs.cs(39,27): warning CS8604: Possible null reference argument for parameter 'value' in 'Input<string>.implicit operator Input<string>(string value)'. [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]
/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/FuncWithAllOptionalInputs.cs(19,177): error CS1061: 'InvokeOptions' does not contain a definition for 'WithVersion' and no accessible extension method 'WithVersion' accepting a first argument of type 'InvokeOptions' could be found (are you missing a using directive or an assembly reference?) [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]
/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/FuncWithAllOptionalInputs.cs(25,178): error CS1061: 'InvokeOptions' does not contain a definition for 'WithVersion' and no accessible extension method 'WithVersion' accepting a first argument of type 'InvokeOptions' could be found (are you missing a using directive or an assembly reference?) [/Users/ianwahbe/go/src/github.com/pulumi/pulumi/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/dotnet/Pulumi.Example.csproj]
    3 Warning(s)
    2 Error(s)

Time Elapsed 00:00:01.63
```

Reply:
I see. `options.WithVersion()` is an extension method defined in a generated `Utilities.cs` within the `Pulumi.Example` namespace. The using directive is necessary to call the extension method.

Looking at the schema for this test makes me question one aspect of it. The package name is `example` and all the resource and type tokens in the schema are correctly prefixed with `example:`. However, the function is prefixed with `mypkg:`:

https://github.com/pulumi/pulumi/blob/5da5cdb9dd807c83ca07e3e983dec94dc4ffec34/pkg/codegen/internal/test/testdata/plain-object-disable-defaults/schema.json#L186-L187

I think we need to change that function from `mypkg::funcWithAllOptionalInputs` to `example::funcWithAllOptionalInputs`.

Reply:
Is `"mypkg::funcWithAllOptionalInputs"` a valid function for this schema? If all names _must_ begin with `"example::"` we should fail to process the schema. Because codegen produced an output, I assumed that the name is valid.